src/test/java/com/spotify/docker/PushMojoTest.java

```diff
@@ -47,4 +47,14 @@ public void testPush() throws Exception {
     verify(docker).push(eq("busybox"), any(AnsiProgressHandler.class));
   }
 
+  public void testPushPrivateRepo() throws Exception {
```

Comment:
I don't think there is a way to do so. The only think you can do is stubbing the maven settings as here:
https://maven.apache.org/plugin-testing/maven-plugin-testing-harness/examples/complex-mojo-parameters.html